src/instrumenter/state_vars.ts

```diff
@@ -42,36 +43,59 @@ export type RHS = Expression | [Expression, number];
  * Given potentially complex assignments involving tuples and function return desugaring, return an
  * iterable of all the primitive assignments happening between. (i.e. assignments where the LHS is not a tuple)
  */
-function* getAssignmentComponents(lhs: Expression, rhs: Expression): Iterable<[LHS, RHS]> {
-    if (lhs instanceof TupleExpression) {
-        if (lhs.vOriginalComponents.length === 1 && !lhs.isInlineArray) {
-            if (lhs.vOriginalComponents[0] !== null) {
-                yield* getAssignmentComponents(lhs.vOriginalComponents[0], rhs);
+function* getAssignmentComponents(
+    lhs: Expression | VariableDeclarationStatement,
+    rhs: Expression
+): Iterable<[LHS, RHS]> {
+    if (lhs instanceof TupleExpression || lhs instanceof VariableDeclarationStatement) {
+        let assignedDecls: Array<null | VariableDeclaration | Expression>;
+
+        if (lhs instanceof TupleExpression) {
+            assignedDecls = lhs.vOriginalComponents;
+        } else {
+            assignedDecls = lhs.assignments.map((declId) =>
+                declId === null ? null : (lhs.requiredContext.locate(declId) as VariableDeclaration)
+            );
+        }
+
+        if (assignedDecls.length === 1) {
+            if (assignedDecls[0] !== null) {
+                yield* getAssignmentComponents(assignedDecls[0], rhs);
             }
         } else if (rhs instanceof TupleExpression) {
```

Comment:
I'm wondering if `rhs` of `TupleExpression` with truthy [`isInlineArray`](https://github.com/ConsenSys/solc-typed-ast/blob/64c1d1e3ee4bba27a25ca46a4e8d9fc494b1bbe5/src/ast/implementation/expression/tuple_expression.ts#L8) may cause trouble. Maybe not, because assigned declarations will still match only whole array literal and not it's parts. But if there would be a way to check such edge-case - it would be cool.

Reply:
So the check is on the `lhs`, and I don't think a LHS expression can ever be an inline array, which is why I removed it